### DISCUSSION TITLE
How to investigate why JobDetails is not cacheable

### DISCUSSION THREAD

**Discussion by kamer** [jobrunr/jobrunr#1123] [Q&A] [answered]
Hello,
I have a simple job as follows:

```java
BackgroundJob.enqueue(() -> someService.doSomething(str1, str2));
```

It runs but I encounter _Job details not cacheable!"_ error on dashboard. I checked the best practices page as error message suggests but cannot see why it's not cacheable. 

Is there a way to debug why a job is not cacheable? I tried to add some debugger breakpoints but cannot find out. (I'm using Micronaut integration with redis-lettuce type if it means something.)

**Comment by rdehuyss** [comment]
Hi @kamer  - this is not an error but a warning and the only downside of this is that creating jobs will be a bit slower.

As soon as you do some string additions or a calculation inside the job lambda you give JobRunr, your job details will not be cacheable. See the example below:
```java 
BackgroundJob.enqueue(() -> someService.doSomething(str1 + " " + str2));
``` 

I just tested the example you gave and that is cacheable on our side. Can you create a reproducer?

**Reply by kamer** [reply]
Hi @rdehuyss ,

Thanks for the answer. Here's the reproducer.

https://github.com/kamer/jobrunnr-reproducer-1123/

I did not use Micronaut integration for the reproducer. You can find running instructions in README.

**Reply by rdehuyss** [reply, marked as answer]
I found the reason - it's because you inject the interface instead of the actual class. It is a valid bug - still need to see how we can fix it.

**Comment by kamer** [comment]
I can confirm it's fixed with v7.3.1. Thanks.
